staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -259,7 +264,16 @@ func (r *proxyHandler) updateAPIService(apiService *apiregistrationv1api.APIServ
 		servicePort:      *apiService.Spec.Service.Port,
 		serviceAvailable: apiregistrationv1apihelper.IsAPIServiceConditionTrue(apiService, apiregistrationv1api.Available),
 	}
-	if r.proxyTransport != nil && r.proxyTransport.DialContext != nil {
+	if r.egressSelector != nil {
+		networkContext := egressselector.Cluster.AsNetworkContext()
+		var egressDialer utilnet.DialFunc
+		egressDialer, err := r.egressSelector.Lookup(networkContext)
+		if err != nil {
+			klog.Warning(err.Error())
+		} else {
+			newInfo.restConfig.Dial = egressDialer
+		}
+	} else if r.proxyTransport != nil && r.proxyTransport.DialContext != nil {
```

Comment:
So, if set, `proxyTransport.DialContext` is no longer used?

Does this change the behavior / meaning of the "--enable-aggregator-routing" flag?

Reply:
proxyTransport.DialContext contains the ssh dial if ssh is enabled. So the logic here says, if egressSelector is enabled, use it to get a dialer, otherwise use the ssh dialer if there is one.

"--enable-aggregator-routing": Turns on aggregator routing requests to endpoints IP rather than cluster IP.

Either IP will be in the cluster network, so I think using the egressSelector doesn't change the behavior.

Reply:
OK thanks.